Jaewon

jwkim395 opened the pull request:
피드백 적용 사항
1. 나이 입력 추가
2. 변수명 변경
3. std::string, std::map 적용

구현사항
1. 이름으로 정보 검색
2. 회원 생성시 중복체크
 - 이름 중복 확인
3. 이름 순으로 정렬해서 출력
 - map이 tree 구조라 정렬이 적용됨
 - 역순은 아직 적용 X

kimjw395 committed:
회원정보 관리프로그램 2주차(1)
피드백 적용 사항
1. 나이 입력 추가
2. 변수명 변경
3. map, string 적용

구현사항
1. 이름으로 정보 검색
2. 회원 생성시 중복체크
 - 이름 중복 확인
3. 이름 순으로 정렬해서 출력
 - map이 tree 구조라 정렬이 적용됨
 - 역순은 아직 적용 X

kimjw395 committed:
회원 정보 역순 출력

kimjw395 committed:
필터링 출력 구현
성별에 따라, 나이대(10대, 20대 등)에 따라 필터링해서 출력

jeontigger commented on Jaewon/Project1/Project1/main.cpp:
```diff
@@ -111,22 +121,137 @@ void findUser()
+	if (reverse) {
+		map<string, UserData>::iterator iter;
+		for (iter = storage.begin(); iter != storage.end(); ++iter) {
+			printf("회원 %d\n", cnt);
+			cout << "이름 : " << iter->first << endl;
+			if (iter->second.sex == 1)
+				printf("성별 : 남자\n");
+			else
+				printf("성별 : 여자\n");
+			printf("나이 : %d\n", iter->second.age);
+			++cnt;
```
반복되는 출력형식 => 함수화 하면 좋을듯!

jeontigger commented on Jaewon/Project1/Project1/main.cpp:
```diff
@@ -111,22 +121,137 @@ void findUser()
+		printf("2. 20대\n");
+		printf("3. 30대\n");
+		printf("4. 40대 이상\n");
+		cin >> sel;
+		int selectedAge;
+		switch (sel)
+		{
+		case 0:
+			selectedAge = 0;
+			break;
+		case 1:
+			selectedAge = 10;
+			break;
+		case 2:
+			selectedAge = 20;
+			break;
+		case 3:
+			selectedAge = 30;
+			break;
+		case 4:
+			selectedAge = 40;
+			break;
+		default:
+			break;
+		} 
```
다른 좋은 방법이 있을 것으로 생각됨!

kimjw395 committed:
피드백 적용
- 반복되는 출력형식 함수화(printData 함수 오버로딩해서 구현)
- selectedAge 형식 변경(switch에서 if문의 조건문으로)

kimjw395 committed:
요구사항 구현
1. 회원 정보 수정
2. 회원 정보 삭제
3. UserData에 group 멤버 변수 추가

jeontigger reviewed (commented):
printData와 switch가 산술연산 한 번으로 변경된 것 굿!
- 성별은 왜 int에서 bool로 변경했는지?
- 동명이인은 중복이 아님!
-  but,  수정과 삭제에서는 동명이인중 한명만 지우면 ok

jwkim395 commented:
bool쪽으로 바꾼 이유는 단순하게 0, 1로 놓고 있기 때문에 !연산자로 쉽게 바꿀 수 있을 것 같아서 바꾸어 봤음
중복은 다음 커밋 때 한번 바꿔볼께
